payment/payment_test.go

```diff
@@ -111,5 +117,59 @@ func TestFromPublicKeys(t *testing.T) {
 	if p2sh != "XJkohBHRMT8JUknSqCH7aJP9gAuAe9eNLY" {
 		t.Errorf("TestScriptHash: error when encoding script hash")
 	}
+}
+
+func TestPaymentConfidentialPubKeyHash(t *testing.T) {
+	expected := "VTpzxkqVGbraaCz18fQ2GxLvZkupCi2MPtUdt9ygAEeZ8v9gZPtkD5RUc" +
+		"ap55WZ3aVsbUG6TsQvXc8R3"
+	pk1 := "030000000000000000000000000000000000000000000000000000000000000001"
+	pk1Byte, err := hex.DecodeString(pk1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pubKey1, err := btcec.ParsePubKey(pk1Byte, btcec.S256())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pk2 := "030000000000000000000000000000000000000000000000000000000000000001"
+	pk2Byte, err := hex.DecodeString(pk2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pubKey2, err := btcec.ParsePubKey(pk2Byte, btcec.S256())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	payment := payment.FromPublicKey(pubKey1, &network.Liquid, pubKey2)
+	assert.Equal(t, expected, payment.ConfidentialPubKeyHash())
+}
+
+func TestPaymentConfidentialScriptHash(t *testing.T) {
+	expected := "VJLCUu2hpcjPaTGMnANXni8wVYjsCAiTEznE5zgRZZyAWXE2P6rz6Dvph" +
+		"BHSn7iz4w9sLb3mFSHGJbte"
+	scriptHash, err := hex.DecodeString(
```

Comment:
we should use a script here instead of its hash. This wrongly led you to use the `Script` instead of the `Hash` into the `ConfidentialScriptHash`, I guess.

From the reference test vector, we should translate the [script](https://github.com/vulpemventures/liquidjs-lib/blob/97673986c4233dee61f98dc59b5256ed2a8f313d/test/fixtures/p2sh.json#L83) in ASM version to the hex encoding, so we can convert it to bytes in the test using `hex`.
The proper script should be:
```
a9149f840a5fc02407ef0ad499c2ec0eb0b942fb008687
```